share/spack/gitlab/cloud_pipelines/stacks/python/spack.yaml

```diff
@@ -0,0 +1,1829 @@
+    - py-unshare
+    - py-update-checker
+    - py-uproot
+    - py-uproot3
+    - py-uproot3-methods
+    - py-uproot4
+    - py-uritemplate
+    - py-urllib3
+    - py-urwid
+    - py-us
+    - py-usgs
+    - py-utils
+    - py-uvloop
+    - py-uvw
+    - py-uwsgi
+    - py-vcf-kit
+    - py-vcrpy
+    - py-vcstool
+    - py-vcstools
+    - py-vcversioner
+    - py-vector-quantize-pytorch
+    - py-vermin
+    - py-versioneer
+    - py-vine
+    - py-virtualenv
+    - py-virtualenv-clone
+    - py-virtualenvwrapper
+    - py-visdom
+    - py-voluptuous
+    - py-vsc-base
+    - py-vsc-install
+    - py-vsts
+    - py-vsts-cd-manager
+    - py-walinuxagent
+    - py-wand
+    - py-wandb
+    - py-warlock
+    - py-warpx
+    - py-wasabi
+    - py-watchdog
+    - py-wcsaxes
+    - py-wcwidth
+    - py-weave
+    - py-webencodings
+    - py-webkit-server
+    - py-weblogo
+    - py-webob
+    - py-websocket-client
+    - py-websockets
+    - py-werkzeug
+    - py-wget
+    - py-whatshap
+    - py-wheel
+    - py-whichcraft
+    - py-whoosh
+    - py-widgetsnbextension
+    - py-wincertstore
+    - py-word2number
+    - py-workload-automation
+    - py-wradlib
+    - py-wrapt
+    - py-wstool
+    - py-wub
+    - py-wurlitzer
+    - py-wxmplot
+    - py-wxpython
+    - py-xarray
+    - py-xattr
+    - py-xdot
+    - py-xenv
+    - py-xgboost
+    - py-xlrd
+    - py-xlsxwriter
+    - py-xlwt
+    - py-xmlrunner
+    - py-xmltodict
+    - py-xopen
+    - py-xpyb
+    - py-xvfbwrapper
+    - py-xxhash
+    - py-yacs
+    - py-yahmm
+    - py-yajl
+    - py-yamlreader
+    - py-yapf
+    - py-yarl
+    - py-yolk3k
+    - py-youtube-dl
+    - py-yq
+    - py-yt
+    - py-ytopt
+    - py-ytopt-autotune
+    - py-zarr
+    - py-zc-buildout
+    - py-zc-lockfile
+    - py-zict
+    - py-zipp
+    - py-zope-event
+    - py-zope-interface
+    - py-zxcvbn
+    - pythia8
+    - qscintilla
+    - rdkit
+    - reditools
+    - redland-bindings
+    - reprimand
+    - saga-gis
+    - scons
+    - sensei
+    - sgpp
+    - simsipm
+    - simulationio
+    - sirius
+    - snakemake
+    - sollve
+    - sumo
+    - sz
+    - talloc
+    - tasmanian
+    - tfel
+    - thrift
+    - timemory
+    - tskit
+    - tulip
+    - uqtk
+    - verrou
+    - vigra
+    - vtk
+    - xcfun
+    - xrootd
+    - xtensor-python
+    - z3
+
+  mirrors: { "mirror": "s3://spack-binaries-develop/python" }
+
+  gitlab-ci:
+
+    script:
+      - . "./share/spack/setup-env.sh"
+      - spack --version
+      - cd ${SPACK_CONCRETE_ENV_DIR}
+      - spack env activate --without-view .
+      - spack config add "config:install_tree:projections:${SPACK_JOB_SPEC_PKG_NAME}:'morepadding/{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}'"
+      - mkdir -p ${SPACK_ARTIFACTS_ROOT}/user_data
+      - spack -d ci rebuild > >(tee ${SPACK_ARTIFACTS_ROOT}/user_data/pipeline_out.txt) 2> >(tee ${SPACK_ARTIFACTS_ROOT}/user_data/pipeline_err.txt >&2)
+
+    mappings: []
```

Comment:
I don't *think* leaving `mappings` as an empty list works, but I could be mistaken.  Once you get the stack to concretize, if you see a bunch of messages like "No match found for {spec}, skipping it", that will be be due to missing any mappings here.  See the list in the `data-vis-sdk` stack, [here](https://github.com/spack/spack/blob/f81d84dfc6b1c3a821dffab785e62ecd79f130ab/share/spack/gitlab/cloud_pipelines/stacks/data-vis-sdk/spack.yaml#L50), for how you can map large builds to more capable instances, and then just have a "catch-all" to map everything else.